tests/Microsoft.Bot.Builder.Tests/TranscriptUtilities.cs

```diff
@@ -24,19 +26,162 @@ public static class TranscriptUtilities
         /// <returns>A list of activities to test</returns>
         public static IEnumerable<IActivity> GetFromTestContext(TestContext context)
         {
-            var transcriptsRootFolder = TestUtilities.GetKey("TranscriptsRootFolder") ?? @"..\..\..\..\..\transcripts";
-            var directory = Path.Combine(transcriptsRootFolder, context.FullyQualifiedTestClassName.Split('.').Last());
-            var fileName = $"{context.TestName}.transcript";
-            var path = Path.Combine(directory, fileName);
+            var relativePath = Path.Combine(context.FullyQualifiedTestClassName.Split('.').Last(), $"{context.TestName}.chat");
+            return GetActivities(relativePath);
+        }
+
+        public static IEnumerable<IActivity> GetActivities(string relativePath)
+        {
+            var transcriptsRootFolder = TranscriptUtilities.EnsureTranscriptsDownload();
+            var path = Path.Combine(transcriptsRootFolder, relativePath);
+            if (!File.Exists(path))
+            {
```

Comment:
Please add in comments showing / explaining what this code is doing. Reading it, it seems weird as the two if statemnts are the same, and it's only upon closer inspection that they're modifying the variable they're testing.